js/private/js_library.bzl

```diff
@@ -88,6 +88,32 @@ runtime dependency on this target.
         providers = [JsInfo],
     ),
     "data": JS_LIBRARY_DATA_ATTR,
+    "no_copy_to_bin": attr.label_list(
+        allow_files = True,
+        doc = """List of files to not copy to the Bazel output tree when `copy_data_to_bin` is True.
+
+        This is useful for exceptional cases where a `copy_to_bin` is not possible or not suitable for an input
+        file such as a file in an external repository. In most cases, this option is not needed.
+        See `copy_data_to_bin` docstring for more info.
+        """,
+    ),
+    "copy_data_to_bin": attr.bool(
+        doc = """When True, `data` files and the `entry_point` file are copied to the Bazel output tree before being passed
+        as inputs to runfiles.
+
+        Defaults to True so that a `js_library` with the default value is compatible with `js_run_binary` with
```

Comment:
2nd and 3rd paragraphs here are js_binary specific and can be dropped. Just the 1st paragraph will suffice